src/experimental/tools/mediaCapabilitiesProber/probers/__tests__/decodingInfo.test.ts

```diff
@@ -52,28 +51,27 @@ describe("MediaCapabilitiesProber probers - decodingInfo", () => {
     navigator.mediaCapabilities = origMediaCapabilities;
   });
 
-  it("should throw if no video and audio config", () => {
+  it("should throw if no video and audio config", async () => {
     const decodingInfoStub = stubDecodingInfo(true);
     const configuration = {
       type: "media-source" as const,
     };
 
-    expect.assertions(2);
-    return probeDecodingInfos(configuration)
-      .then(() => {
-        resetDecodingInfos();
-      })
-      .catch(({ message }: { message: string }) => {
-        expect(message).toEqual(
-          "MediaCapabilitiesProber >>> API_CALL: " +
-            "Not enough arguments for calling mediaCapabilites.",
-        );
-        expect(decodingInfoStub).not.toHaveBeenCalled();
-        resetDecodingInfos();
-      });
+    let thrownException = false;
+    try {
+      await probeDecodingInfos(configuration);
+      resetDecodingInfos();
+    } catch (error) {
+      const message = error instanceof Error ? error.message : "";
+      thrownException = true;
+      expect(message).toEqual("Not enough arguments for calling mediaCapabilites.");
+      expect(decodingInfoStub).not.toHaveBeenCalled();
+      resetDecodingInfos();
+    }
```

Comment:
remark about `resetDecodingInfos()`,  it can be done only once outside the try..catch

```js
try {
  await probeDecodingInfos(configuration);
} catch (error) {
  const message = error instanceof Error ? error.message : "";
  thrownException = true;
  expect(message).toEqual("Not enough arguments for calling mediaCapabilites.");
  expect(decodingInfoStub).not.toHaveBeenCalled();
}
resetDecodingInfos();
```